lib/easy.c

```diff
@@ -89,6 +89,55 @@
 static unsigned int  initialized;
 static long          init_flags;
 
+#if defined(USE_THREADS_POSIX) || (defined(USE_THREADS_WIN32) && _WIN32_WINNT >= 0x0600)
```

Comment:
Note that these `USE_THREADS_` defines are for using the threaded name resolvers. Meaning that builds that for example use c-ares for DNS don't have those defines set.

Reply:
Thanks for the review.

Would it be possible to enable pthread and win32 threads by default or via another configure option (like --init-thread-safe)?

By the way, this PR introduces a leak of the CRITICAL_SECTION on Windows (we can't clean it from curl_global_cleanup since this would prevent a future curl_global_init to work).

A potential alternative would be to implement a mutex using the "Futuxes Are Tricky" paper from Drepper (and use WIN32 WaitOnAddress() as futuxes) but this also requires adding atomics...